### PR TITLE
Fix dependabot alerts

### PR DESCRIPTION
**Pull request by kuritka**
- Fix dependabot issues
- bump go to 1.17
- bump several package versions
related to #28

To upgrade to GO1.17 I had to make changes in Github pipes
 - Change Lint, Terratest jobs execution on push
 - Upgrade linter to v1.45.0
 - Upgrade Go1.17 for terratest
 - bump k3d-action to @v2 (newest)

Signed-off-by: kuritka <kuritka@gmail.com>

### CODE REVIEW

**Review by k0da — APPROVED** [review]

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -1,7 +1,12 @@
 name: Lint
 
 on:
-  pull_request:
```
Comment (jkremser) [inline .github/workflows/lint.yml:4]: why not running the linter for PRs?

File: .github/workflows/terratest.yaml [hunk 2]
```diff
@@ -1,7 +1,12 @@
 name: Terratest
 
 on:
-  pull_request:
```
Comment (jkremser) [inline .github/workflows/terratest.yaml:4]: same here, imho the PRs should be the first class citizens here

**Review by jkremser — CHANGES_REQUESTED** [review]
otherwise lgtm 👍

File: .github/workflows/lint.yml (hunk 1 above)
Comment (kuritka) [reply .github/workflows/lint.yml:4]: fixed

Comment (jkremser) [reply .github/workflows/lint.yml:4]: thanks

**Review by jkremser — APPROVED** [review]

**Pull request merged**